fix: synchronize Resource and ResourceRef API (#2309)
kyma-project/busola: docs/extensibility/form-section.md

  - **version** - _[required]_ API version used for all requests.
  - **scope** - either `namespace` or `cluster`. When set to `cluster`, namespaced resources are fetched from all Namespaces. Defaults to `cluster`.
  - **namespace** - Namespace to fetch resources from. Used only when scope is `namespace` and resources need to be fetched from a specific Namespace. Defaults to the active Namespace when omitted.

> - **filter** - jsonata expression used to filter resources by the given condition.

Reviewer: ```suggestion
- **filter** - JSONata expression that filters resources based on a given condition.
```

> - **provideVar** - when this field is defined, the chosen resource will be provided as a variable of this name.

Reviewer: ```suggestion
- **provideVar** - when this field is defined, the chosen resource is provided as a variable of this name.
```

  - **kind** - _[required]_ Kubernetes kind of the resource.
  - **group** - API group used for all requests. Not provided for Kubernetes resources in the core (also called legacy) group.
  - **version** - _[required]_ API version used for all requests.

> - **filter** - jsonata expression used to filter resources by the given condition.

Reviewer: ```suggestion
- **filter** - JSONata expression that filters resources based on a given condition.
```
